exporters/otlp/src/recordable.cc

```diff
@@ -132,15 +132,20 @@ void Recordable::AddEvent(nostd::string_view name,
   });
 }
 
-void Recordable::AddLink(opentelemetry::trace::SpanContext span_context,
+void Recordable::AddLink(const opentelemetry::trace::SpanContext &span_context,
                          const common::KeyValueIterable &attributes) noexcept
 {
   auto *link = span_.add_links();
   attributes.ForEachKeyValue([&](nostd::string_view key, common::AttributeValue value) noexcept {
     PopulateAttribute(link->add_attributes(), key, value);
     return true;
   });
-  // TODO: Populate trace_id, span_id and trace_state when these are supported by SpanContext
+
+  span_.set_trace_id(reinterpret_cast<const char *>(span_context.trace_id().Id().data()),
```

Comment:
I might be missing something - why would we put the link's trace id as the span's trace id?

Reply:
That's a good question. I thought Recordable is meant to capture the link `SpanContext`. Seems I am wrong and we don't need capture the link `SpanContext` here?

Reply:
I think each link has to hold the spancontext data plus a set of attributes.
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#specifying-links

Reply:
This seems wrong. Adding a link should never overwrite the span's trace id.

Can you please add unit tests for this, so we can more better verify those conditions?